### DISCUSSION TITLE
Articulation joint type of cylindrical

### DISCUSSION THREAD

**Discussion by yananZZZZ** [NVIDIA-Omniverse/PhysX#284] [Q&A]
Hi NIVIDA teams,

Currently I want to implement a cylindrical type of articulation joint among adjacent links. however, I found out there is not a such articulation joint type. So, i create a D6 joint with these two links with twist and X axis free. however, it seems displace these links and they are not moving. 

I also tried to use "PxArticulationJointReducedCoordinate::setMotion" to make my articulation joint twist and x axis free. And leave the Joint type unset. it's not working neither. 

I wonder is there any reason cylindrical motion is not support in the articulation joint. any workaround to achieve it? What's more, if I want to implement other joint behavior such as planer, how should I do that? Is there any example code shows how to expand the articulation joint behavior?

Thank you.

**Comment by vreutskyy** [comment]
Hi @yananZZZZ,

PhysX articulations only support sliders (prismatic - 1 linear DOF), hinges (revolute - 1 angular DOF), and ball-socket (spherical - 2 or 3 angular DOFs) joints. To achieve 1 angular and 1 linear DOF between two articulation links, you'll need to add an intermediate link. This intermediate link can have zero length, no collision shape, and be practically invisible. Here’s how you can set it up:

1. **First Link**: The base link.
2. **Second Link**: The intermediate link attached to the first link, which can rotate around a certain axis.
3. **Third Link**: The full-size link attached to the second link, which can slide along the same axis.

This setup will give you the desired 1 angular and 1 linear DOFs.

Hope this helps!
